segmentation_labeling_app/rois/rois.py

```diff
@@ -24,10 +25,112 @@
 sql_insert_roi = """ INSERT INTO rois_prelabeling (transform_hash,
                      ophys_segmentation_commit_hash,
                      creation_date, upload_date, manifest, experiment_id,
-                     roi_id) 
+                     roi_id)
                      VALUES (?, ?, ?, ?, ?, ?, ?) """
 
 
+def cumulative_fraction_threshold(data, target_fraction, nbins=200):
+    """return a threshold value above which the summed weights are
+    target_fraction of the total summed weights
+
+    Parameters
+    ----------
+    data : numpy.ndarray
+        1d array of data, all values > 0
+    target_fraction : float
+        in range [0, 1.0]
+
+    Returns
+    -------
+    fval : float
+        data[data > fval].sum() / data.sum() ~ target_fraction
+
+    """
+    def fopt(x):
+        frac = data[data > x].sum() / data.sum()
+        diff = target_fraction - frac
+        return diff
+
+    fval = bisect(fopt, data.min(), data.max())
```

Comment:
The [docs](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.bisect.html) say this can return a root results object when full_output is true, would it beneficial to have that for testing for convergence and not assuming convergence.

Reply:
It is definitely possible we're going to run into some weird cases. But, we just want to get something running for UI/UX testing. Let's come back and handle weird cases later.

Reply:
What's the reason for this method vs. just computing quantiles (e.g. https://docs.scipy.org/doc/numpy/reference/generated/numpy.quantile.html)

Reply:
No, that's pretty much what I was trying to do, I just didn't know the `quantile` function

Reply:
Thanks, that really cleans things up. Redoing the plots in the PR comments after this change.

Reply:
PR comment updated.